src/mono/mono/mini/aot-compiler.c

```diff
@@ -3764,9 +3764,11 @@ encode_signature (MonoAotCompile *acfg, MonoMethodSignature *sig, guint8 *buf, g
 	if (sig->pinvoke)
 		flags |= 0x80;
 	flags |= (sig->call_convention & 0x0F);
-	// TODO: Encode ext_callconv
-
 	*p = flags;
+
+	++p;
+	*p = sig->ext_callconv & 0x0F;
```

Comment:
Since this is rarely used, it would be better to only emit it if needed, like by setting a bit in 'flags' if its set.

Reply:
The `flags` appear to be fully utilized, but we can add a new variable. In terms of performance, would it be faster to always encode it rather than inserting a single bit and then having a logic for conditional decoding within the runtime?

Reply:
You can make the flags into a guint32, and encode and decode them using encode_value () / decode_value ().
Switch explicit_this to be encoded as 0x80 and pinvoke as 0x40, since pinvoke is more common, then encode having the ext callconv as 0x100. encode_value () will still use 1 byte for values <= 127.

Reply:
Thanks, added.